Sources/momento/utils/CollectionTtl.swift

```diff
@@ -0,0 +1,96 @@
+/**
+ * Represents the desired behavior for managing the TTL on collection
+ * objects (dictionaries, lists, sets) in your cache.
+ *
+ * For cache operations that modify a collection, there are a few things
+ * to consider.  The first time the collection is created, we need to
+ * set a TTL on it.  For subsequent operations that modify the collection
+ * you may choose to update the TTL in order to prolong the life of the
+ * cached collection object, or you may choose to leave the TTL unmodified
+ * in order to ensure that the collection expires at the original TTL.
+ *
+ * The default behavior is to refresh the TTL (to prolong the life of the
+ * collection) each time it is written.  This behavior can be modified
+ * by calling withNoRefreshTtlOnUpdates.
+ */
+public class CollectionTtl {
+    private var _ttlSeconds: Int?
```

Comment:
Can we go ahead and use `TimeInterval` here as well for consistency with the scalar ttls?